File: 2_C++(VisualStudio)/2_workspace_MFC_Direct2D/1_SideViewGameSample/src/src/BlockObject.cpp
#include "pch.h"
#include "include/BlockObject.h"
#include "GameLibrary/GameLibrary.h"
using namespace Library;

TileType BlockObject::GetTileType() {

    return TileType::Ground;
}

void BlockObject::Draw(Renderer& renderer, int x, int y,float offsetY, const Camera& camera){

    GameLibrary::DrawBlock(renderer, x, y, offsetY,camera, TILE_SIZE, ColorF::Brown);
}

File: 99_AI生成ソフトウェア(バイブコーディング)/1_Webアプリ系(VSCode)/3_SideViewGameSample_TypeScript_移植元CPlus(Claude)/SideViewGameSample_移植元CPlus/src/GameLogic/GameObject/BlockObject.ts
import { Renderer } from "../Renderer";
import { Camera } from "../Camera";
import { TileType } from "../TileType";
import { ColorF } from "../ColorF";
import { TileMapObject } from "./TileMapObject";
import { GameLibrary } from "../../GameLibrary/GameLibrary";

/**
 * BlockObject.ts
 * 移植元: include/BlockObject.h / src/BlockObject.cpp
 *
 * 地面ブロック(茶色の疑似立方体)。TileMapObjectを継承し、
 * GetTileType/Draw のみをオーバーライドする。
 */
export class BlockObject extends TileMapObject {

    public override getTileType(): TileType {
        return TileType.Ground;
    }

    public override draw(renderer: Renderer, x: number, y: number, offsetY: number, camera: Camera): void {
        GameLibrary.drawBlock(renderer, x, y, offsetY, camera, TileMapObject.TILE_SIZE, ColorF.Brown);
    }
}
